# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/rocm-llvm/hotfix-glibc-2.42.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- llvm-project-rocm-7.1.1/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp.vanilla	2025-12-20 16:33:05.171407004 +0100
+++ llvm-project-rocm-7.1.1/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp	2025-12-20 16:33:21.139628020 +0100
@@ -457,7 +457,7 @@
 
   // ioctl arguments
   unsigned struct_ifreq_sz = sizeof(struct ifreq);
-  unsigned struct_termios_sz = sizeof(struct termios);
+  //unsigned struct_termios_sz = sizeof(struct termios);
   unsigned struct_winsize_sz = sizeof(struct winsize);
 
 #if SANITIZER_LINUX
@@ -486,7 +486,7 @@
   unsigned struct_mtpos_sz = sizeof(struct mtpos);
   unsigned struct_rtentry_sz = sizeof(struct rtentry);
 #if SANITIZER_GLIBC || SANITIZER_ANDROID
-  unsigned struct_termio_sz = sizeof(struct termio);
+  //unsigned struct_termio_sz = sizeof(struct termio);
 #endif
   unsigned struct_vt_consize_sz = sizeof(struct vt_consize);
   unsigned struct_vt_sizes_sz = sizeof(struct vt_sizes);
@@ -767,8 +767,8 @@
   unsigned IOCTL_SOUND_PCM_WRITE_FILTER = SOUND_PCM_WRITE_FILTER;
 #endif // SOUND_VERSION
   unsigned IOCTL_TCFLSH = TCFLSH;
-  unsigned IOCTL_TCGETA = TCGETA;
-  unsigned IOCTL_TCGETS = TCGETS;
+  //unsigned IOCTL_TCGETA = TCGETA;
+  //unsigned IOCTL_TCGETS = TCGETS;
   unsigned IOCTL_TCSBRK = TCSBRK;
   unsigned IOCTL_TCSBRKP = TCSBRKP;
   unsigned IOCTL_TCSETA = TCSETA;
--- llvm-project-rocm-7.1.1/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp.vanilla	2025-12-20 17:24:18.333464172 +0100
+++ llvm-project-rocm-7.1.1/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp	2025-12-20 17:28:09.473498289 +0100
@@ -48,7 +48,7 @@
 #include <sys/times.h>
 #include <sys/types.h>
 #include <sys/utsname.h>
-#include <termios.h>
+#include <term.h>
 #include <time.h>
 #include <wchar.h>
 #include <regex.h>
@@ -768,12 +768,12 @@
 #endif // SOUND_VERSION
   unsigned IOCTL_TCFLSH = TCFLSH;
   //unsigned IOCTL_TCGETA = TCGETA;
-  //unsigned IOCTL_TCGETS = TCGETS;
+  unsigned IOCTL_TCGETS = TCGETS;
   unsigned IOCTL_TCSBRK = TCSBRK;
   unsigned IOCTL_TCSBRKP = TCSBRKP;
-  unsigned IOCTL_TCSETA = TCSETA;
-  unsigned IOCTL_TCSETAF = TCSETAF;
-  unsigned IOCTL_TCSETAW = TCSETAW;
+  //unsigned IOCTL_TCSETA = TCSETA;
+  //unsigned IOCTL_TCSETAF = TCSETAF;
+  //unsigned IOCTL_TCSETAW = TCSETAW;
   unsigned IOCTL_TCSETS = TCSETS;
   unsigned IOCTL_TCSETSF = TCSETSF;
   unsigned IOCTL_TCSETSW = TCSETSW;
--- llvm-project-rocm-7.1.1/compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors_ioctl.inc.vanilla	2025-12-20 17:32:07.521456654 +0100
+++ llvm-project-rocm-7.1.1/compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors_ioctl.inc	2025-12-20 17:34:47.037343138 +0100
@@ -339,21 +339,21 @@
   _(SOUND_PCM_WRITE_FILTER, WRITE, sizeof(int));
   _(TCFLSH, NONE, 0);
 #if SANITIZER_GLIBC
-  _(TCGETA, WRITE, struct_termio_sz);
+  //_(TCGETA, WRITE, struct_termio_sz);
 #endif
-  _(TCGETS, WRITE, struct_termios_sz);
+  //_(TCGETS, WRITE, struct_termios_sz);
   _(TCSBRK, NONE, 0);
   _(TCSBRKP, NONE, 0);
 #if SANITIZER_GLIBC
-  _(TCSETA, READ, struct_termio_sz);
-  _(TCSETAF, READ, struct_termio_sz);
-  _(TCSETAW, READ, struct_termio_sz);
+  //_(TCSETA, READ, struct_termio_sz);
+  //_(TCSETAF, READ, struct_termio_sz);
+  //_(TCSETAW, READ, struct_mermio_sz);
 #endif
   _(TCSETS, READ, struct_termios_sz);
   _(TCSETSF, READ, struct_termios_sz);
   _(TCSETSW, READ, struct_termios_sz);
   _(TCXONC, NONE, 0);
-  _(TIOCGLCKTRMIOS, WRITE, struct_termios_sz);
+  //_(TIOCGLCKTRMIOS, WRITE, struct_termios_sz);
   _(TIOCGSOFTCAR, WRITE, sizeof(int));
   _(TIOCINQ, WRITE, sizeof(int));
   _(TIOCLINUX, READ, sizeof(char));
@@ -361,7 +361,7 @@
   _(TIOCSERGETLSR, WRITE, sizeof(int));
   _(TIOCSERGWILD, WRITE, sizeof(int));
   _(TIOCSERSWILD, READ, sizeof(int));
-  _(TIOCSLCKTRMIOS, READ, struct_termios_sz);
+  //_(TIOCSLCKTRMIOS, READ, struct_termios_sz);
   _(TIOCSSOFTCAR, READ, sizeof(int));
   _(VT_ACTIVATE, NONE, 0);
   _(VT_DISALLOCATE, NONE, 0);
--- llvm-project-rocm-7.1.1/compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors.inc.vanilla	2025-12-20 17:51:50.784209928 +0100
+++ llvm-project-rocm-7.1.1/compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors.inc	2025-12-20 17:53:32.180751944 +0100
@@ -3933,7 +3933,7 @@
 #define INIT_WCTOMB
 #endif
 
-#if SANITIZER_INTERCEPT_TCGETATTR
+#if 0 //SANITIZER_INTERCEPT_TCGETATTR
 INTERCEPTOR(int, tcgetattr, int fd, void *termios_p) {
   void *ctx;
   COMMON_INTERCEPTOR_ENTER(ctx, tcgetattr, fd, termios_p);
--- llvm-project-rocm-7.1.1/compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors_ioctl.inc.vanilla	2025-12-20 17:54:52.948030728 +0100
+++ llvm-project-rocm-7.1.1/compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors_ioctl.inc	2025-12-20 17:56:00.796169490 +0100
@@ -349,10 +349,10 @@
   //_(TCSETAF, READ, struct_termio_sz);
   //_(TCSETAW, READ, struct_mermio_sz);
 #endif
-  _(TCSETS, READ, struct_termios_sz);
-  _(TCSETSF, READ, struct_termios_sz);
-  _(TCSETSW, READ, struct_termios_sz);
-  _(TCXONC, NONE, 0);
+  //_(TCSETS, READ, struct_termios_sz);
+  //_(TCSETSF, READ, struct_termios_sz);
+  //_(TCSETSW, READ, struct_termios_sz);
+  //_(TCXONC, NONE, 0);
   //_(TIOCGLCKTRMIOS, WRITE, struct_termios_sz);
   _(TIOCGSOFTCAR, WRITE, sizeof(int));
   _(TIOCINQ, WRITE, sizeof(int));
